rollup.config.js

```diff
@@ -29,12 +29,17 @@ let lessPlugin = less({
     }
 });
 
+let intro = `window.addEventListener('WebComponentsReady', function () {`;
+let outro = `});`;
```

Comment:
just making sure I understand this correctly but this is added basically to wrap the bundle in the event listener /polyfill correct?  I assume this is for browsers that need the polyfill?

Reply:
## TL;DR

Without this fix, Firefox will throw a  `TypeError: Constructor HTMLElement requires 'new'` error. I ran across this issue when prototyping early 2017 and the only way to fix it was to wrap the class definition in a `WebComponentsReady` event listener. 

## Long-winded Explanation
The best I can narrow it down to is that this happens when trying to instantiate any custom element with a shadow dom before the ES5 adapter has had a chance to polyfill the API. 

This doesn't always happen, though. On a fresh load in Firefox (stale cache or something), it seems to work fine, but any subsequent load throws the error. 

This leads me to conclude that there's a difference in Firefox's loading algorithm between an initial load and subsequent loads, resulting in a race condition that is rarely won.

The `intro/outro` options basically wrap the compiled ES5 class definitions in their own event listener to ensure that things load correctly.

_Note:_ if we didn't have to support IE, we wouldn't have to compile to ES5, we wouldn't need the adapter, and we wouldn't need the extra code to make it work. This fix only applies to ES5-compiled javascript.

Reply:
Good find and thanks for the detailed explanation